src/main/java/com/gmail/inverseconduit/chat/StackExchangeChat.java

```diff
@@ -52,19 +74,96 @@ public StackExchangeChat() {
+                    return false;
+                String response = getRes.getContentAsString();
+                LOGGER.info("Got response from fetching fkey of OpenID login:" + response);
+
+                WebRequest post = new WebRequest(
+                    UrlUtils.toUrlUnsafe("https://openid.stackexchange.com/account/login/submit"),
+                    HttpMethod.POST
+                );
+                try {
+                    post.setRequestParameters(ImmutableList.of(
+                        new NameValuePair("email", email),
+                        new NameValuePair("password", password),
+                        new NameValuePair("fkey",
+                            builder.parse(getRes.getContentAsStream())
+                                .getElementById("fkey")
+                                .getAttribute("value")
+                        )
+                    ));
+                } catch(SAXException die) {
+                    // throw new IllegalStateException("Parsing response text as DOM has caused a parse exception.", die);
+                    return false;
+                }
+                WebResponse postRes = webClient.loadWebResponse(post);
+                if (null == postRes || !Strings.isNullOrEmpty(postRes.getResponseHeaderValue("p3p")))
+                    // throw new IllegalArgumentException("Could not post to submit of OpenID.");
+                    return false;
+            }
+
+            { // Log into Stack Exchange Chat.
+                WebResponse getRes = webClient.loadWebResponse(new WebRequest(
+                    UrlUtils.toUrlUnsafe("http://stackexchange.com/users/chat-login"),
+                    HttpMethod.GET
+                ));
+                WebRequest post = new WebRequest(
+                    UrlUtils.toUrlUnsafe("http://chat.stackexchange.com/users/login/global"),
+                    HttpMethod.POST
+                );
+                try {
+                    Document dom = builder.parse(getRes.getContentAsStream());
+                    post.setRequestParameters(ImmutableList.of(
+                        new NameValuePair("authToken", dom.getElementById("authToken").getAttribute("value")),
+                        new NameValuePair("nonce", dom.getElementById("nonce").getAttribute("value"))
+                    ));
+                } catch(SAXException die) {
+                    // throw new IllegalStateException("Parsing response text as DOM has caused a parse exception.", die);
+                    return false;
+                }
+                post.setAdditionalHeaders(ImmutableMap.of(
+                    "Origin", "http://chat.stackexchange.com",
+                    "Referer", "http://chat.stackexchange.com"
+                ));
+                WebResponse postRes = webClient.loadWebResponse(post);
+                String response = postRes.getContentAsString();
+                return response.contains("Welcome");
```

Comment:
This seems extremely brittle to me. Probably a failed login gets us a http status code 403 instead. That would *probably* be better